### DISCUSSION TITLE
Boost.Multiprecision support

### DISCUSSION THREAD

**Discussion by zxffffffff** [nlohmann/json#4480] [General]
The precision issues with `float` and `double` have long been a challenging aspect of programming, particularly when JSON is used for network transmission, Java's `BigDecimal` often gets converted to `number`, but there isn't a well-suited solution on the C++ side.

I've typically used a workaround involving `json_dump() -> parse_string() -> cpp_dec_float_50`, which is rather inelegant. It would be ideal if a JSON library, known for its ease of use, could natively support this.

**Comment by nlohmann** [comment]
Looking into this.

(Also linking this to our existing tracking item for user-defined/custom number types, since arbitrary-precision support like Boost.Multiprecision would build on the same underlying mechanism.)

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*

**Reply by zxffffffff** [reply]
Thank you, Claude code. I've been waiting for a long time. ♥️
